modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -1301,15 +1396,35 @@
         breakout-field-mappings (into {} (map (juxt identity field-alias)) breakout)
         ;; We have already sorted ascending by the breakout fields so we don't have to repeat the
         ;; same sort.
-        explicit-order-by (and (seq order-by)
-                               (not= order-by (map (fn [field] [:asc field]) breakout)))]
-    (binding [*field-mappings* (merge *field-mappings* breakout-field-mappings)]
-      (cond-> pipeline-ctx
-        (seq sort-fields) (update :query conj
-                                  ;; We $addFields before sorting, otherwise expressions will not be available for the sort
-                                  {$addFields (into (ordered-map/ordered-map) sort-fields)})
-        explicit-order-by (update :query conj
-                                  (order-by->$sort order-by))))))
+        explicit-order-by
+        (when (and (seq order-by)
+                   (not= order-by (map (fn [field] [:asc field]) breakout)))
+          (binding [*field-mappings* (merge *field-mappings* breakout-field-mappings)]
+            (order-by->$sort order-by)))
+
+        cumulative-order-by
+        (when-let [finest-temporal-index
+                   (and (seq (filter #(#{:cum-sum :cum-count} (first (second %))) aggregation))
+                        (qp.util.transformations.nest-breakouts/finest-temporal-breakout-index breakout 2))]
+          (let [id (projection-group-map breakout)]
+            (as-> id lst
+              (m/remove-nth finest-temporal-index lst)
+              (mapv (fn [[name]] [name 1]) lst)
+              (conj lst [(first (nth (seq id) finest-temporal-index)) 1])
+              (filter (fn [[key]] (not (and explicit-order-by
+                                            (explicit-order-by key)))) lst))))
+
+        combined-order-by
+        (when (or explicit-order-by cumulative-order-by)
+          {$sort (into (ordered-map/ordered-map)
+                       (concat explicit-order-by cumulative-order-by))})]
+    (cond-> pipeline-ctx
+      (seq sort-fields) (update :query conj
+                                ;; We $addFields before sorting, otherwise expressions will not be available for the sort
+                                {$addFields (into (ordered-map/ordered-map) sort-fields)})
+      combined-order-by (update :query #(conj % combined-order-by)))))
+
+;;; ----------------------------------------------------- fields -----------------------------------------------------
```

Comment:
I'm guessing that remove-parent-fields was moved up from here-ish to above the order-by section at some point when the order-by code needed it.  Presumably, the "--- fields ---" label was moved up with remove-parent-fields.   However, that seems wrong, because these labels are generally associated with the matching handle-* function.  Since I was also moving a helper function around, I decided to "fix" this.

If we actually want it the other way, I'm completely fine with moving this label back and (presumably) moving the "--- order-by ---" up above order-by->$sort's new location.